### ISSUE TITLE
v12: Configurator doesn't allow automatic configuration of the DB

### ISSUE THREAD

**Issue by julsemaan** [inverse-inc/packetfence#7119]
**Describe the bug**
On a fresh install, you go in the configurator and when you reach step 2, the database doesn't allow automatic configuration and forces you to do it manually although the DB is virgin

**To Reproduce**
1. Fresh install on Debian 11

**Expected behavior**
Should be allowed to use the automatic configuration of the DB

**Comment by julsemaan** [comment]
Actually I can't configure the DB at all. It requires me to enter a root password, I enter one, hit verify and it refreshes the page

![image](https://user-images.githubusercontent.com/3857942/185473705-ef65ff82-eaa8-4a01-bfd6-4ae797d9b2ab.png)

**Comment by julsemaan** [comment]
Reopening until fix is 100% validated

**Comment by nqb** [comment]
I tested configurator on EL8 against latest `devel` packages and this issue is still here.

In my web browser console, I didn't see any API calls related to database except:

`POST https://172.18.200.10:1443/api/v1/configurator/system_service/packetfence-mariadb/start`

which received a 202 answer.

@satkunas, I think there is still an issue on frontend to fix

**Comment by nqb** [comment]
The behavior I got once I enter a password in Root Password field: redirect to first step of configurator.

If I then go back to second step of configurator, now Automatic configuration is enabled and it works as expected.

**Comment by satkunas** [comment]
@nqb I wasn't able to reproduce using fresh rocky8 install.

**Comment by satkunas** [comment]
I did a 2nd fresh install to verify and was still not able to reproduce. Using Rocky8.

**Comment by nqb** [comment]
Self-assigning, I will retest

**Comment by nqb** [comment]
@satkunas did a new test on Debian 11 without issue. Closing for now.

**Comment by julsemaan** [comment]
I've hit this issue on a fresh Debian 11 install again

It seems like its on/off but it definitely happens

**Comment by julsemaan** [comment]
@satkunas, could you try to dig into why this is happening from time to time and please test using Debian 11 to make sure we're as close as we can to the issue

**Comment by julsemaan** [comment]
Let's note that after retrying the process 3 times, it eventually got the auto-configuration so it definitely seems like an on/off problem

**Comment by satkunas** [comment]
I'm still trying to reproduce on debian.

As a note for what's expected, there's 3 consecutive API calls:

* `POST /api/v1/configurator/config/base/database/test {username: "root"}` => `200 {"message":"Successfully connected to the database mysql with user root","status":200}`
*  `POST /api/v1/configurator/config/base/database/test {username: "root", database: "pf"}` => `500 {"message":"Error in connection to the database pf with user root","status":500}`
* `POST /api/v1/configurator/config/base/database/test {username: "pf", password: "", database: "pf"}` => `500 {"message":"Error in connection to the database pf with user pf","status":500}`

If you're sending different requests, or receiving different responses then please dump them here.

**Comment by julsemaan** [comment]
Can we completely block the form until all API calls are completed (and thus the right state is known).

Looks like right now the database fields are editable although the information hasn't been gathered

**Comment by nqb** [comment]
As I mentioned above, I didnt see any of these API calls when I got this issue.

**Comment by julsemaan** [comment]
I tested this multiple times yesterday and found that the 'poll' after the 'start' can take up to 10 seconds to return. 

Summary of the solution discussed with @satkunas 

When it takes a long time to poll, the DB fields are left editable and you have no feedback that its attempting to detect if auto-config is possible. The solution is to block the database form while its computing if auto-config can be used (15 seconds total at most)

I was able to replicate the issue a few times but it does require a lot of attempts (happens 1 out of 10 times approx). Independently from fixing the form, we need to investigate why the poll is taking that much time in some cases but that doesn't change the need from fixing the form and ensuring the feedback is correct in the frontend.

The frontend fix should be a good-enough fix for 12.0 since it does always work when you wait long enough (~15 seconds at most in my tests)
